ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -223,7 +223,8 @@ private QueryMetadata buildPlanForStructuredOutputNode(
               outputNode.getKsqlTopic(),
               outputNode.getId().toString()
                   + ksqlConfig.getString(KsqlConfig.KSQL_TABLE_STATESTORE_NAME_SUFFIX_CONFIG),
-              schemaKTable.isWindowed()
+              schemaKTable.isWindowed(),
+              true
```

Comment:
How do we know that this topic was created by KSQL?

Reply:
This is the sink stream/table that is created for a query.

Reply:
So we know this is a _sink_ topic, but we don't know if the topic was created by KSQL. The topic could already exist and have data in it, right?

Reply:
Yes, we know that KSQL writes the query results into this topic but it's possible the topic exists already.

Reply:
So `createdByKsql` may be `true`, when actually the topic was _not_ created by KSQL....

Reply:
Correct. We already have the same case when we drop a stream/table along with deleting the corresponding topic. For instance when you drop a stream along with deleting the topic we will delete the topic even though it may have existed when we created the stream.

Reply:
IMHO we should not be adding a flag called `createdByKsql` which can be `true` when the topic may _not_ have been created by KSQL.

Additionally, IMHO, we should _not_ be deleting a topic based on this flag, as it might mean KSQL deletes a topic it didn't create.

However, if I understand the use-case correctly here this change is related to the clean up code. This is to differentiate between source topics declared using CT/CS statements and sink topics declared using CTAS/CSAS/SI statements, right?  

I'm assuming this is needed so that we can return a list of sink topics the user may want to delete as part of the clean up, right?  (The user then choosing which, if any, they want to delete...)

So maybe we're only disagreeing over the name of the variable here. `createdByKsql` is misleading. The topic may _not_ have been created by Ksql. Maybe this is more a flag to indicate the topic is used as a sink topic by KSQL?  So maybe calling it `sinkTopic` would be more correct?

Reply:
Note that this refers to the STREAM/TABLE and not the topic. So `createdByKSQL` makes sense since the STREAM or TABLE is created by KSQL.